Bottom: 1622a70dcb0ee968dd8c0302d5f9516c76b7f27c
Top:    9deb6bf070013a255417da1b2c82dcdfb075aff8
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-11-19 23:01:48 +0000

HID: Merge {fill/thindraw}_pcb_* APIs into HID_DRAW_API structure

I wanted to keep these separate from the pure graphics APIs, but
after poking at this more and more, it seems hard to justify why
we should do that.


---

diff --git a/src/hid.h b/src/hid.h
index d7974e5..6135750 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -282,6 +282,15 @@ typedef enum
     void (*fill_polygon) (hidGC gc, int n_coords_, Coord *x, Coord *y);
     void (*fill_rect)    (hidGC gc, Coord x1, Coord y1, Coord x2, Coord y2);
 
+    /* The following APIs render using PCB data-structures, not immediate parameters */
+
+    void (*fill_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+    void (*thindraw_pcb_polygon) (hidGC gc, PolygonType *poly, const BoxType *clip_box);
+    void (*fill_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
+    void (*thindraw_pcb_pad) (hidGC gc, PadType *pad, bool clip, bool mask);
+    void (*fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
+    void (*thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
+
   } HID_DRAW_API;
 
 
@@ -372,16 +381,6 @@ typedef enum
 
     HID_DRAW_API *graphics;
 
-    void (*fill_pcb_polygon) (hidGC gc_, PolygonType *poly,
-                              const BoxType *clip_box);
-    void (*thindraw_pcb_polygon) (hidGC gc_, PolygonType *poly,
-                                  const BoxType *clip_box);
-    void (*fill_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
-    void (*thindraw_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
-    void (*fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
-    void (*thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
-
-
     /* This is for the printer.  If you call this for the GUI, xval and
        yval are ignored, and a dialog pops up to lead you through the
        calibration procedure.  For the printer, if xval and yval are
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index f49bc19..0644f00 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -332,7 +332,7 @@ hid_batch_init ()
   memset (&batch_hid, 0, sizeof (HID));
 
   common_nogui_init (&batch_hid);
-  common_draw_helpers_init (&batch_hid);
+  common_draw_helpers_init (&batch_graphics);
 
   batch_hid.struct_size           = sizeof (HID);
   batch_hid.name                  = "batch";
diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 1a9cdb2..82d7955 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -169,7 +169,7 @@ hid_extents_init (void)
   memset (&extents_hid, 0, sizeof (HID));
   memset (&extents_graphics, 0, sizeof (HID_DRAW_API));
 
-  common_draw_helpers_init (&extents_hid);
+  common_draw_helpers_init (&extents_graphics);
 
   extents_hid.struct_size         = sizeof (HID);
   extents_hid.name                = "extents-extents";
diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 4466529..952621f 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -1578,7 +1578,7 @@ hid_gcode_init ()
   memset (&gcode_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&gcode_hid);
-  common_draw_helpers_init (&gcode_hid);
+  common_draw_helpers_init (&gcode_graphics);
 
   gcode_hid.struct_size         = sizeof (HID);
   gcode_hid.name                = "gcode";
diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index b884c51..5f5aebc 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -1269,7 +1269,7 @@ hid_gerber_init ()
   memset (&gerber_graphics, 0, sizeof (gerber_graphics));
 
   common_nogui_init (&gerber_hid);
-  common_draw_helpers_init (&gerber_hid);
+  common_draw_helpers_init (&gerber_graphics);
 
   gerber_hid.struct_size         = sizeof (gerber_hid);
   gerber_hid.name                = "gerber";
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 1913210..665a1a4 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -34,6 +34,7 @@
 #endif
 
 extern HID ghid_hid;
+extern HID_DRAW_API ghid_graphics;
 
 static hidGC current_gc = NULL;
 
@@ -789,8 +790,8 @@ ghid_init_renderer (int *argc, char ***argv, GHidPort *port)
 
   /* Setup HID function pointers specific to the GL renderer*/
   ghid_hid.end_layer = ghid_end_layer;
-  ghid_hid.fill_pcb_polygon = ghid_fill_pcb_polygon;
-  ghid_hid.thindraw_pcb_polygon = ghid_thindraw_pcb_polygon;
+  ghid_graphics.fill_pcb_polygon = ghid_fill_pcb_polygon;
+  ghid_graphics.thindraw_pcb_polygon = ghid_thindraw_pcb_polygon;
 }
 
 void
@@ -880,7 +881,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
      we can't use the hidgl polygon drawing routine */
   /* TODO: We could use the GLU tessellator though */
   if (hidgl_stencil_bits() == 0)
-    ghid_hid.fill_pcb_polygon = common_fill_pcb_polygon;
+    ghid_graphics.fill_pcb_polygon = common_fill_pcb_polygon;
 
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 4e8f0fe..61b1070 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2087,7 +2087,7 @@ REGISTER_FLAGS (ghid_main_flag_list)
 #endif
 
 HID ghid_hid;
-static HID_DRAW_API ghid_graphics;
+HID_DRAW_API ghid_graphics;
 
 void
 hid_gtk_init ()
@@ -2130,7 +2130,6 @@ hid_gtk_init ()
   memset (&ghid_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&ghid_hid);
-  common_draw_helpers_init (&ghid_hid);
 
   ghid_hid.struct_size              = sizeof (HID);
   ghid_hid.name                     = "gtk";
@@ -2183,6 +2182,8 @@ hid_gtk_init ()
 
   ghid_hid.graphics                 = &ghid_graphics;
 
+  common_draw_helpers_init (&ghid_graphics);
+
   ghid_graphics.make_gc             = ghid_make_gc;
   ghid_graphics.destroy_gc          = ghid_destroy_gc;
   ghid_graphics.use_mask            = ghid_use_mask;
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 5261ee4..94441fe 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -4043,7 +4043,6 @@ hid_lesstif_init ()
   memset (&lesstif_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&lesstif_hid);
-  common_draw_helpers_init (&lesstif_hid);
 
   lesstif_hid.struct_size             = sizeof (HID);
   lesstif_hid.name                    = "lesstif";
@@ -4092,6 +4091,8 @@ hid_lesstif_init ()
 
   lesstif_hid.graphics                = &lesstif_graphics;
 
+  common_draw_helpers_init (&lesstif_graphics);
+
   lesstif_graphics.make_gc             = lesstif_make_gc;
   lesstif_graphics.destroy_gc          = lesstif_destroy_gc;
   lesstif_graphics.use_mask            = lesstif_use_mask;
diff --git a/src/hid/lpr/lpr.c b/src/hid/lpr/lpr.c
index b33773e..7aecc51 100644
--- a/src/hid/lpr/lpr.c
+++ b/src/hid/lpr/lpr.c
@@ -129,6 +129,7 @@ hid_lpr_init ()
   memset (&lpr_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&lpr_hid);
+  common_draw_helpers_init (&lpr_graphics);
   ps_ps_init (&lpr_hid);
   ps_ps_graphics_init (&lpr_graphics);
 
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index dfaf3f5..6c837be 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -1040,7 +1040,7 @@ hid_nelma_init()
   memset (&nelma_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&nelma_hid);
-  common_draw_helpers_init (&nelma_hid);
+  common_draw_helpers_init (&nelma_graphics);
 
   nelma_hid.struct_size         = sizeof (HID);
   nelma_hid.name                = "nelma";
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 848bac7..62cb62f 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -1806,7 +1806,7 @@ hid_png_init ()
   memset (&png_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&png_hid);
-  common_draw_helpers_init (&png_hid);
+  common_draw_helpers_init (&png_graphics);
 
   png_hid.struct_size = sizeof (HID);
   png_hid.name        = "png";
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index f09c601..1a43106 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -670,7 +670,7 @@ hid_eps_init ()
   memset (&eps_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&eps_hid);
-  common_draw_helpers_init (&eps_hid);
+  common_draw_helpers_init (&eps_graphics);
 
   eps_hid.struct_size         = sizeof (HID);
   eps_hid.name                = "eps";
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index d665e6c..376be9c 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1526,7 +1526,7 @@ hid_ps_init ()
   memset (&ps_graphics, 0, sizeof (HID_DRAW_API));
 
   common_nogui_init (&ps_hid);
-  common_draw_helpers_init (&ps_hid);
+  common_draw_helpers_init (&ps_graphics);
   ps_ps_init (&ps_hid);
   ps_ps_graphics_init (&ps_graphics);
